transport/http_util.go

```diff
@@ -247,9 +279,14 @@ func (d *decodeState) addMetadata(k, v string) {
 func (d *decodeState) processHeaderField(f hpack.HeaderField) error {
 	switch f.Name {
 	case "content-type":
-		if !validContentType(f.Value) {
+		contentSubtype, validContentType := getContentSubtype(f.Value)
+		if !validContentType {
 			return streamErrorf(codes.FailedPrecondition, "transport: received the unexpected content-type %q", f.Value)
 		}
+		d.contentSubtype = contentSubtype
+		// TODO: do we want to propagate the whole content-type, or some up with
+		// a way to just propagate the content-subtype if it was set?
```

Comment:
Maybe I don't understand...  Do you mean "strip off application/grpc"?  If so, I think I like this more (propagate the whole value verbatim).

Reply:
I added some clarification